File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int, int> m;
        for (int i = 0; i < n; i++) {
            int diff = target - nums[i];
            if (m.count(diff)) return {m[diff], i};
            else m[nums[i]] = i;
        }
        return {-1, -1};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        mp={}
        for i in range(len(numbers)):
            t = target-numbers[i]
            if t in mp:
                return [mp[t],i+1]
            mp[numbers[i]]=i+1
        return [-1,-1]